src/main/java/ch/heigvd/dai/network/SocketClient.java

```diff
@@ -0,0 +1,158 @@
+/*
+ * Wheel Of Fortune - a Java server/client implementation of the television game
+ * Copyright (C) 2024 Pedro Alves da Silva, Gonçalo Carvalheiro Heleno
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+
+package ch.heigvd.dai.network;
+
+import com.google.common.net.HostAndPort;
+import java.io.*;
+import java.net.*;
+import java.net.UnknownHostException;
+import java.nio.charset.StandardCharsets;
+
+public class SocketClient extends SocketAbstract {
+  public enum ClientCommand {
+    HELLO,
+    INVALID,
+    HELP,
+    QUIT
+  }
+
+  public enum ServerCommand {
+    HELLO_ACK,
+    INVALID
+  }
+
+  public SocketClient(HostAndPort hostAndPort)
+      throws NullPointerException, IllegalArgumentException, UnknownHostException {
+    super(hostAndPort);
+  }
+
+  @Override
+  public void run() {
+    System.out.println(
+        // TODO Check if the output includes brackets when connecting to IPv6
+        "[Client] Connecting to server at " + getHost().getHostAddress() + ":" + getPort() + "...");
+
+    try (Socket socket = new Socket(getHost(), getPort());
+        Reader reader = new InputStreamReader(socket.getInputStream(), StandardCharsets.UTF_8);
+        BufferedReader in = new BufferedReader(reader);
+        Writer writer = new OutputStreamWriter(socket.getOutputStream(), StandardCharsets.UTF_8);
+        BufferedWriter out = new BufferedWriter(writer)) {
+      // Print message to acknowledge successful connection to the server.
+      // TODO Check if the output includes brackets when connecting to IPv6
+      System.out.println(
+          "[Client] Connected to server at " + getHost().getHostAddress() + ":" + getPort());
+      System.out.println();
+
+      // Print help message.
+      help();
+
+      // Run REPL until user quits with command QUIT.
+      while (!socket.isClosed()) {
+        // Output prompt.
+        System.out.print("> ");
+
+        // Read user input from console.
+        Reader inputReader = new InputStreamReader(System.in, StandardCharsets.UTF_8);
+        BufferedReader bir = new BufferedReader(inputReader);
+        String userInput = bir.readLine();
+
+        try {
```

Comment:
Use try with ressources? If you don't want to change these, `inputReader` and `bir` are never closed

Reply:
Done! I've moved them into the same try-with-resources we use to open other readers.